inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -203,18 +206,42 @@ void Engine::AddExtension(InferenceEngine::IExtensionPtr extension) {
 }
 
 void Engine::QueryNetwork(const ICNNNetwork& network, const std::map<std::string, std::string>& config, QueryNetworkResult& res) const {
-    details::CNNNetworkIterator i(&network);
-    while (i != details::CNNNetworkIterator()) {
-        try {
-            mkldnn::engine eng(mkldnn::engine(mkldnn::engine::kind::cpu, 0));
-            MKLDNNWeightsSharing::Ptr fake_w_cache;
-
-            // if we can create and have not thrown exception, then layer is supported
-            std::unique_ptr <MKLDNNNode>(MKLDNNNode::CreateNode(*i, eng, extensionManager, fake_w_cache));
-            res.supportedLayersMap.insert({ (*i)->name, GetName() });
-        } catch (InferenceEngine::details::InferenceEngineException&) {
+    MKLDNNWeightsSharing::Ptr fake_w_cache;
+    auto function = network.getFunction();
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        for (auto&& node : function->get_ops()) {
+            if (!node->is_constant() && !node->is_parameter() && !node->is_output()) {
```

Comment:
Does it mean that all Constants (regardless of type and shape) and parameters (any kind of, even future extensions, like Parameters with default values or something like this; regardless of type and shape) are treated as always supported?

Reply:
No. This means that we do not take this nodes into account. It is supposed that they ofloaded to the same plugin as operation node with which they connected. I don't think that we have support of subgraph that conssists of single constant or parameter node in any plugin.